pkg/kernel/kernel.go

```diff
@@ -211,6 +243,13 @@ func (k *kernel) Running() <-chan struct{} {
 	return k.running
 }
 
+func (k *kernel) exit() {
+	k.exitOnce.Do(func() {
+		k.logger.Info("leaving kernel with exit code %d", k.exitCode)
+		k.exitHandler(k.exitCode)
```

Comment:
If I set an empty exit handler, will this then automatically always return an exit code of 0?

Reply:
Leaving an empty exit handler will give back control to the caller..  
In our case this would lead to an exit code of 0, yes